public/game/actions.js

```diff
@@ -397,8 +397,32 @@ function move(state, {move}) {
  * @returns state
  */
 function dealDamageEqualToBlock(state, {target}) {
-	const block = state.player.block
-	return removeHealth(state, {target, amount: block})
+	if (state.player.block) {
+		const block = state.player.block
+		return removeHealth(state, { target, amount: block })
+	}
+}
+
+function dealDamageEqualToVulnerable(state, { target }) {
+	return produce(state, (draft) => {
+		getTargets(draft, target).forEach((t) => {
+			if (t.powers.vulnerable) {
+				t.currentHealth = t.currentHealth - t.powers.vulnerable
```

Comment:
I sometimes have (had?) weird bug when updating health so I recommend always using the action instead of directly. Although some times it does work fine.

```suggestion
				removeHealth(draft, {target: t, amount: t.currentHealth - t.powers.vulnerable})
```